x/treasury/internal/keeper/indicator.go

```diff
@@ -11,104 +12,89 @@ import (
+// - SR: Computes the Seigniorage Reward
+// - TRL: Computes the Tax Reward per unit Luna
 // - SMR: Computes the ratio of seigniorage rewards to overall mining rewards
 //
 // Rolling averages are also computed for MRL and SMR respectively.
 //
 
-// TaxRewardsForEpoch returns tax rewards that have been collected in the epoch
-func TaxRewardsForEpoch(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	taxRewards := sdk.NewDecCoins(k.PeekTaxProceeds(ctx, epoch))
-
-	taxRewardInMicroSDR := sdk.ZeroDec()
-	for _, coinReward := range taxRewards {
-		if coinReward.Denom != core.MicroSDRDenom {
-			swappedReward, err := k.marketKeeper.ComputeInternalSwap(ctx, coinReward, core.MicroSDRDenom)
+// alignCoins align the coins to the given denom through the market swap
+func (k Keeper) alignCoins(ctx sdk.Context, coins sdk.DecCoins, denom string) (alignedAmt sdk.Dec) {
+	alignedAmt = sdk.ZeroDec()
+	for _, coinReward := range coins {
+		if coinReward.Denom != denom {
+			swappedReward, err := k.marketKeeper.ComputeInternalSwap(ctx, coinReward, denom)
 			if err != nil {
 				continue
 			}
-			taxRewardInMicroSDR = taxRewardInMicroSDR.Add(swappedReward.Amount)
+			alignedAmt = alignedAmt.Add(swappedReward.Amount)
 		} else {
-			taxRewardInMicroSDR = taxRewardInMicroSDR.Add(coinReward.Amount)
+			alignedAmt = alignedAmt.Add(coinReward.Amount)
 		}
 	}
 
-	return taxRewardInMicroSDR
+	return alignedAmt
 }
 
-// SeigniorageRewardsForEpoch returns seigniorage rewards for the epoch
-func SeigniorageRewardsForEpoch(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	seignioragePool := k.PeekEpochSeigniorage(ctx, epoch)
-	rewardAmt := k.GetRewardWeight(ctx, epoch).MulInt(seignioragePool)
-	seigniorageReward := sdk.NewDecCoinFromDec(core.MicroLunaDenom, rewardAmt)
+// UpdateIndicators updates interal indicators
+func (k Keeper) UpdateIndicators(ctx sdk.Context) {
+	epoch := core.GetEpoch(ctx)
+	lunaTotalBondedAmount := k.stakingKeeper.TotalBondedTokens(ctx)
 
-	microSDRReward, err := k.marketKeeper.ComputeInternalSwap(ctx, seigniorageReward, core.MicroSDRDenom)
-	if err != nil {
-		return sdk.ZeroDec()
-	}
+	// Compute Tax Rewards & Tax Rewards per (staked)Luna
+	taxRewards := sdk.NewDecCoins(k.PeekTaxProceeds(ctx))
+	TR := k.alignCoins(ctx, taxRewards, core.MicroSDRDenom)
+	TRL := TR.QuoInt(lunaTotalBondedAmount)
 
-	return microSDRReward.Amount
-}
+	k.SetTRL(ctx, epoch, TRL)
 
-// MiningRewardForEpoch returns the sum of tax and seigniorage rewards for the epoch
-func MiningRewardForEpoch(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	taxRewards := TaxRewardsForEpoch(ctx, k, epoch)
-	seigniorageRewards := SeigniorageRewardsForEpoch(ctx, k, epoch)
+	// Reset tax proceeds after computing TRL for the next epoch
+	k.SetTaxProceeds(ctx, sdk.Coins{})
 
-	return taxRewards.Add(seigniorageRewards)
-}
+	// Compute Seigniorage Rewards
+	seigniorage := k.PeekEpochSeigniorage(ctx)
+	seigniorageRewardsAmt := k.GetRewardWeight(ctx).MulInt(seigniorage)
```

Comment:
Isn't the seigniorage rewards = oracle reward pool / oracle reward distribution period? 

Not all seigniorage created in an epoch gets distributed during the epoch; it could theoretically get amortized over a much longer period depending on how the oracle reward distribution period is set

Reply:
@YunSuk-Yeo did we think about this?